addons/analytic/static/src/js/analytic_distribution/analytic_distribution.js

```diff
@@ -0,0 +1,606 @@
+                break;
+            }
+        }
+        return description;
+    }
+
+    groupStatus(id) {
+        let group = this.list[id];
+        let ready_tags = this.listReadyByGroup(id);
+        if (group.distribution.length > ready_tags.length) {
+            if (group.applicability === 'mandatory') return 'orange'
+            return 'gray';
+        }
+        let sum = this.sumByGroup(id);
+        if (group.applicability === 'mandatory' && (sum < 99.99 || sum >= 100.01)){
+            return 'red';
+        }
+        return 'green';
+    }
+
+    listReadyByGroup(id) {
+        return this.list[id].distribution.filter((tag) => this.tagIsReady(tag));
+    }
+
+    tagIsReady({analytic_account_id, percentage}) {
+        return !!analytic_account_id && !!percentage;
+    }
+
+    sumByGroup(id) {
+        return this.listReadyByGroup(id).reduce((prev, next) => prev + (parseFloat(next.percentage) || 0), 0);
+    }
+
+    remainderByGroup(id) {
+        return 100 - Math.min(this.sumByGroup(id), 100);
+    }
+
+    newTag(group_id) {
+        return {
+            id: this.nextId++,
+            group_id: group_id,
+            analytic_account_id: null,
+            analytic_account_name: "",
+            percentage: this.remainderByGroup(group_id),
+            color: this.list[group_id].color,
+        }
+    }
+
+    autoFill() {
+        for (let group of this.plans.filter((p) => p.account_count > 0)){
+            if (this.remainderByGroup(group.id)) {
+                console.log('autoFill adding lines');
+                this.addLineToGroup(group.id);
+            }
+        }
+    }
+
+    cleanUp() {
+        for (let group_id in this.list){
+            this.list[group_id].distribution = this.listReadyByGroup(group_id);
+        }
+    }
+
+    validate() {
+        for (let group_id in this.list) {
+            if (this.groupStatus(group_id) === 'red') {
+                this.invalidate();
+                console.log('invalidate called because ', group_id, ' is ', this.groupStatus(group_id));
+                return false;
+            }
+        }
+        return true;
+    }
+
+    invalidate() {
+        this.props.record.setInvalidField(this.props.name);
+    }
+
+    async save() {
+        console.log('saving locally')
+        const currentDistribution = this.listForJson;
+        const dataToSave = currentDistribution;
+        await this.props.update(dataToSave);
+        this.validate();
+    }
+
+    onSaveNew() {
+        this.openTemplate({ resId: false, context: {
+            'default_analytic_distribution': this.listForJson,
+            'default_name': this.listReady.map((dist_tag) => dist_tag.analytic_account_name).join(),
+        }})
+    }
+
+    resetRecentlyClosed() {
+        this.recentlyClosed = false;
+    }
+
+    forceCloseEditor() {
+        // this is used when focus will remain/return to the main Element but the dropdown should not open
+        this.recentlyClosed = true;
```

Comment:
Shouldn't this be in the setup, defaulting to false with a comment explaining what it is used for? (like most other class variables)